File: api/src/functions/translate.js
module.exports = async function (context, req) {
    context.log('JavaScript HTTP trigger function processed a request.');

    const api = 'https://api-free.deepl.com/v2/translate';
    try {
            const params = new URLSearchParams();
            params.append('text', req.value);
            params.append('source_lang', 'JA');
            params.append('target_lang', 'EN');
    
            const response = await fetch(api, {
            method: 'POST',
            mode: 'cors',
            headers: {
                'Authorization': `DeepL-Auth-Key ${process.env["DEEPL_API_KEY"]}`
            },
            body: params
        });

        if (response.ok) {
            const json = await response.json();
            context.log(`${json}`);

            context.res.json(json);
        } else {
            const json = await response.json();
            context.log(`${json}`);

            context.res.json(json);
        }
    } catch(e) {
        context.log(`${e}`);
 
        context.res.json({
            error: e
        });
    }
}

File: api/Translate.cs
using System;
using System.IO;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Net.Http;
using System.Collections.Generic;
using Microsoft.Extensions.Configuration;
using System.Text.Json;

namespace WebSpeechAPITest;

public class Translate
{
    private readonly HttpClient _httpClient;
    private readonly ILogger<Translate> _logger;
    private readonly IConfiguration _configuration;

    public Translate(
        IHttpClientFactory httpClientFactory,
        ILogger<Translate> logger,
        IConfiguration configuration
    )
    {
        _httpClient = httpClientFactory.CreateClient();
        _logger = logger;
        _configuration = configuration;
    }

    [FunctionName("translate")]
    public async Task<IActionResult> Run(
        [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req
    )
    {
        var param = JsonSerializer.Deserialize<Dictionary<string, string>>(req.Body);
        _logger.LogInformation($"{param}");

        var api = "https://api-free.deepl.com/v2/translate";

        var content = new FormUrlEncodedContent(param);

        var request = new HttpRequestMessage(HttpMethod.Post, api);
        request.Content = content;
        request.Headers.Add("Authorization", $"DeepL-Auth-Key ${_configuration["DEEPL_API_KEY"]}");

        var response = await _httpClient.SendAsync(request);

        var responseMessage = await response.Content.ReadAsStringAsync();

        return new OkObjectResult(responseMessage);
    }
}
